crates/services/txpool/src/txpool.rs

```diff
@@ -371,19 +361,6 @@ where
             return Err(Error::NoMetadata)
         }
 
-        // verify max gas is less than block limit
-        let block_gas_limit = self
-            .config
-            .chain_config
-            .consensus_parameters
-            .block_gas_limit();
-        if tx.max_gas() > block_gas_limit {
```

Comment:
The conversion into a `Checked` transaction verifies that `max_gas` is less than `tx_max_gas`.